presto-ranger-plugin/src/main/java/io/prestosql/ranger/PrestoAuthorizer.java

```diff
@@ -0,0 +1,182 @@
+    {
+        List<RangerPrestoResource> rangerPrestoResources = new ArrayList<>();
+        for (RangerPrestoResource rangerPrestoResource : resources) {
+            if (checkPermisionForResource(rangerPrestoResource, identity, PrestoAccessType.USE)) {
+                rangerPrestoResources.add(rangerPrestoResource);
+            }
+        }
+        return rangerPrestoResources;
+    }
+
+    public boolean canSeeResource(RangerPrestoResource resource, Identity identity)
+    {
+        return checkPermisionForResource(resource, identity, PrestoAccessType.USE);
+    }
+
+    public boolean canCreateResource(RangerPrestoResource resource, Identity identity)
+    {
+        return checkPermisionForResource(resource, identity, PrestoAccessType.CREATE);
+    }
+
+    public boolean canDropResource(RangerPrestoResource resource, Identity identity)
+    {
+        return checkPermisionForResource(resource, identity, PrestoAccessType.DROP);
+    }
+
+    public boolean canUpdateResource(RangerPrestoResource resource, Identity identity)
+    {
+        return checkPermisionForResource(resource, identity, PrestoAccessType.UPDATE);
+    }
+
+    public boolean canSelectResource(RangerPrestoResource resource, Identity identity)
+    {
+        return checkPermisionForResource(resource, identity, PrestoAccessType.SELECT);
+    }
+
+    private boolean checkPermisionForResource(RangerPrestoResource resource, Identity identity, PrestoAccessType prestoAccessType)
+    {
+        Optional<RangerAccessResult> rangerPrestoPlugin = checkPermission(resource, identity, prestoAccessType);
+        if (rangerPrestoPlugin.isPresent()) {
+            if (resource.getSchemaTable().isPresent() && INFORMATION_SCHEMA_NAME.equals(resource.getSchemaTable().get().getSchemaName())) {
+                return true;
+            }
+            return rangerPrestoPlugin.get().getIsAllowed();
+        }
+        else {
+            return true;
+        }
+    }
+
+    private Optional<RangerAccessResult> checkPermission(RangerPrestoResource resource, Identity identity, PrestoAccessType accessType)
+    {
+        Optional<RangerPrestoPlugin> rangerPrestoPlugin = catalogPlugin.getPluginForCatalog(resource.getCatalogName());
+        if (rangerPrestoPlugin.isPresent()) {
+            RangerPrestoAccessRequest rangerRequest = new RangerPrestoAccessRequest(
+                    resource,
+                    identity.getUser(),
+                    getGroups(identity),
+                    accessType);
+
+            return Optional.of(rangerPrestoPlugin.get().isAccessAllowed(rangerRequest));
+        }
+        return Optional.empty();
+    }
+
+    private Set<String> getGroups(Identity identity)
+    {
+        return userGroups.getUserGroups(identity.getUser());
+    }
+
+    public String getRowLevelFilterExp(String catalogName, RangerPrestoResource resource, Identity identity)
```

Comment:
The row filter methods should be removed, they arent being used right now.

Reply:
column mask changes too

Reply:
Acked